test/dtypes/test_affine_quantized.py

```diff
@@ -36,8 +36,14 @@ def get_quantization_functions(do_sparse: bool, do_int4: bool, device: str = "cu
             base_functions.append(
                 int4_weight_only(group_size=32, layout=Int4CPULayout())
             )
+            if int4_zp_int:
+                base_functions.append(
+                    int4_weight_only(group_size=32, layout=Int4CPULayout(), zero_point_domain=ZeroPointDomain.INT)
```

Comment:
can you add a new section in https://github.com/pytorch/ao/blob/main/torchao/quantization/README.md#other-available-quantization-techniques to report the e2e model level numbers for int4wo cpu?

Reply:
The PR focus on the exposure of the API. The first backend to really run the models still depends on https://github.com/pytorch/pytorch/pull/137566, in which we register ```_weight_int4pack_mm_with_scale_and_zeros``` and make zeros not the same dtype of scales available in PyTorch.  cc @liangan1 here for awareness

However, if you prefer the 2 changes (zero_point_domain as arguments, separate scales and zeros tensors for int4_mm) in 1 PR, I am fine too.

Reply:
oh I see, so this is for xpu, I'd suggest to add a new layout for xpu and configure everything through that layout for now, instead of exposing these options to user, something like this:
```
def int4_weight_only(...., layout):
        if isinstance(layout, Int4XPULayout):
            # set zero_point_domain and other settings here
            ...
```

does this work for your use case?

Reply:
I'd put it not only for XPU but for all backends. CPU and CUDA side might also want integer zero points, as what other popular recipes do. For XPU we are also working on floating zero points,

Do you prefer I binding the integer zp to XPU in this PR temporarily?

Reply:
oh OK, if you want to support both, then it makes sense to expose I think, just make sure to assert in places when it's not supported